rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -250,6 +250,14 @@ impl PendingOperation for PendingMessage {
             return PendingOperationResult::Drop;
         }
 
+        // Perform a preflight check to see if we can short circuit the gas
+        // payment requirement check early without performing expensive
+        // operations like metadata building or gas estimation.
+        if let Err(op_result) = self.meets_gas_payment_requirement_preflight_check().await {
```

Comment:
noting this will slightly increase the duration of time spent in the `prepare` step

Reply:
yeah I think this is worth it - I'll actually add some screenshots to the description to give some more background on the impact of this